build.gradle

```diff
@@ -149,7 +149,51 @@ publishing {
 
 bintrayUpload.dependsOn build
 
+task aggregatedJavadocs(type: Javadoc,
+		description: "Generate javadocs from all child projects as if it was a single project",
+		group: "Documentation"
+){
+	def destination = file("${rootDir}/docs/javadoc/${rootProject.version.toLowerCase()}")
+
+	doFirst {
+		// replace existing directory
+		if (destination.exists()) {
+			destination.deleteDir()
+		}
+	}
+
+	destinationDir destination
+	failOnError false
+
+	options {
+        encoding = 'UTF-8'
+        tags = ["apiNote:a:API Note:",
+                "implSpec:a:Implementation Requirements:",
+                "implNote:a:Implementation Note:"]
+		title = "$rootProject.name ${rootProject.version} API"
+		windowTitle = "$rootProject.name ($version)"
+        addStringOption "encoding", "UTF-8"
+		addStringOption "Xdoclint:none", "-quiet"
+		author true
+		// adding links to javadocs for filling specific classes when not existing inside current javadoc
+		links = [
+				"http://projectreactor.io/docs/core/release/api/",
+				"http://projectreactor.io/docs/netty/release/api/",
+				"http://projectreactor.io/docs/extra/release/api/"
+		]
+	}
+
+	subprojects.each { project ->
+		project.tasks.withType(Javadoc.class).each { doc ->
+			source += doc.source
+			classpath += doc.classpath
+			excludes += doc.excludes
+			includes += doc.includes
+		}
+	}
+}
+
```

Comment:
It would create conflict with other packages?

Reply:
maybe you know what you do?

Reply:
No conflict since we are running this task only on root project which doesn't have a javadoc yet. We need it to conform to the "javadoc" classifier because that way it can be picked up by jitpack.